### PR TITLE
Refactor and fix tractor spawning

### LINKED ISSUE #52: Deck tractor sometimes not starting
The deck tractor seems to be randomly broken. Spawning in the right place, looking good, 0 damage, but when the player tries to drive it, the engine doesn't turn on.

### PR DESCRIPTION
**Pull request by ImplicitDeny**
Basically the whole file have changed. Brought to modern standards.

Tractor's engine on initial spawn was flooded because getPos. I hate getPos. Really. Now it is spawning inside the ship's belly, which is damn useful because it is very slightly above water, always predictable (whereas 0,0,0 is not), and for some reason contain a geometry LOD that doesn't make any sense but allow to walk inside the hull. 🤷 I'll probably use this tip to rework some spawning stuff in the mod.

Also I'm tired and need to go to the foodstore to buy a few things, so have fun reviewing this, for now I'm done. See'ya

PS : fix #52 
Love closing words, but always forget to use them 🤷

### CODE REVIEW

File: addons/gsri_opex/functions/fremm/fn_fremmAddHeli.sqf [hunk 1]
```diff
@@ -1,133 +1,129 @@
...
-	private _condition = {
-		// params["_target", "_player", "_args"];
-		params["", "", "_args"];
-		_args params ["_ship"];
-		// Return true if there is a heli in hangar and if there is no heli on deck
-		!(isNull ([(_ship getVariable "GSRI_FREMM_hangar")] call GSRI_fnc_heliRetrieveCurrent)) and (isNull ([(_ship getVariable "GSRI_FREMM_deck")] call GSRI_fnc_heliRetrieveCurrent))
+// Clientside-only : basically ACE actions
+if!(isDedicated) then {
+	// Actions handle spawn
+	private _handle = ((_ship getVariable "GSRI_FREMM_screens") select 0);
```
Comment (ArwynFr) [inline addons/gsri_opex/functions/fremm/fn_fremmAddHeli.sqf:33]: don't we have a function for this in OPEX ?

**Review by ArwynFr — COMMENTED** [review]
Up to 4 levels of indentation is a lot IMO. Also a lot of comments describing the next bloc of features is a good hint a file has multiple responsibilities and shall be split into smaller chunks. Many `if(condition) { }` blocks ought to be replaced by functions with `if!(condition) { return }` statements.

File: addons/gsri_opex/functions/fremm/fn_fremmAddHeli.sqf (hunk 1 above)
Comment (ImplicitDeny) [reply addons/gsri_opex/functions/fremm/fn_fremmAddHeli.sqf:33]: Thinking of this https://github.com/team-gsri/GSRI-Opex/blob/master/addons/gsri_opex/functions/screen/fn_screenGetById.sqf i guess ? I'll switch to that when back to home

**Comment by ImplicitDeny** [comment]
> Up to 4 levels of indentation is a lot IMO. Also a lot of comments describing the next bloc of features is a good hint a file has multiple responsibilities and shall be split into smaller chunks. Many `if(condition) { }` blocks ought to be replaced by functions with `if!(condition) { return }` statements.

Most of the fremmAdd* files are following this structure (which is basically two if- blocks, one for serverside stuff and one for clientside stuff executed after the server init is finished and the clientCanLoad flag published). We could split all of them in one server-dedicated function and one client-dedicated function, however I'm unsure of the utility of this.

I nonetheless think that this file is getting a bit too big... I'll see what we can do to reduce/split it

**Comment by ImplicitDeny** [comment]
Actually, I don't think it's indented that much. The worst case is a if- block in a loop (given that we keep the two if- general structure). The "fourth level" of indentation is actually a code block definition, not executed. I'm quite unsure of the utility of changing for a splitted server and client files. That will still bring us to a three level indentation, the only level removed being the easiest one to read. Getting rid of more levels will only lead to a shitload of very small files (only a few lines) for each condition and/or statement of ACEs actions.

**Comment by ArwynFr** [comment]
I think the file mixes 4 concerns : tug, helicopter, fuel pump and fries

**Review by ArwynFr — APPROVED** [review]

**Pull request merged**